### PR TITLE
*: remove the LightningMode from TiDB

### PR DESCRIPTION
**Pull request by lonng**
Signed-off-by: Lonng <heng@lonng.org>

### What problem does this PR solve? 

The lightning does not use the TiDB's `kvencoder` anymore due to performance issue.

### What is changed and how it works?

1. Remove the lightning mode from TiDB.
2. Remove the `kvencoder` from TiDB.

### Check List 

Tests 

 - Unit test
 - Integration test

Code changes

 - Has exported variable/fields change

Release note

 - No need to add to release note.

### CODE REVIEW

**Comment by lonng** [comment]
@kennytm PTAL

**Comment by ngaut** [comment]
Nice clean up.

File: session/session.go [hunk 1]
```diff
@@ -246,10 +246,6 @@ func (s *session) DDLOwnerChecker() owner.DDLOwnerChecker {
 }
 
 func (s *session) getMembufCap() int {
-	if s.sessionVars.LightningMode {
-		return kv.ImportingTxnMembufCap
```
Comment (kennytm) [inline session/session.go]: Could the constant `ImportingTxnMembufCap` be removed as well?

Comment (lonng) [reply session/session.go]: Nice catch.

**Comment by lonng** [comment]
@kennytm It's OK to remove the flag from TiDB-Lightning.

**Pull request merged**